cpp/src/arrow/python/arrow_to_python.cc

```diff
@@ -121,6 +126,12 @@ Status GetValue(PyObject* context, const Array& arr, int64_t index, int32_t type
       *result = PyUnicode_FromStringAndSize(reinterpret_cast<const char*>(str), nchars);
       return CheckPyError();
     }
+    case Type::HALF_FLOAT: {
+      uint16_t halffloat = static_cast<const HalfFloatArray&>(arr).Value(index);
+      *result = PyArrayScalar_New(Half);
+      PyArrayScalar_ASSIGN(*result, Half, halffloat);
```

Comment:
Can either of these function calls fail?

Reply:
@wesm `PyArrayScalar_New` returns `NULL` when it fails `malloc`. So we have to `NULL`-check before calling `PyArrayScalar_ASSIGN`.

Reply:
When `*result == NULL`, which status code should I return?

Reply:
You can use `RETURN_IF_PYERROR();`, since a failure should set a Python exception